api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -232,17 +233,38 @@ abstract class WorksController[M <: MultipleResultsRequest[W],
     * of the form /works/{id}.
     *
     */
-  private def respondWithRedirect(originalUri: String,
+  private def respondWithRedirect(originalRequest: Request,
                                   work: IdentifiedRedirectedWork,
-                                  contextUri: String) =
+                                  contextUri: String) = {
+
+    val path = originalRequest.path.replaceAll(
+      s"/${work.canonicalId}",
+      s"/${work.redirect.canonicalId}"
+    )
+
+    val params = originalRequest.params.toMap
+      .filterNot {
+        case (k, _) => k == "id"
```

Comment:
From reading this is the URL `/works/{id}` converted to `/works?id=10`?

Reply:
If you don't filter this param you can go from a search -> redirected item -> actual item with ?id=previous_id, which seems undesirable.

Reply:
I didn't know we had an id param.